iib/workers/tasks/utils.py

```diff
@@ -280,3 +282,32 @@ def _get_function_arg_value(arg_name, func, args, kwargs):
     if arg_value is None and len(args) > arg_index:
         arg_value = args[arg_index]
     return arg_value
+
+
+def get_running_subprocess_output(stream):
+    """
+    Get output of a running process in a non-blocking way.
+
+    :param TextIOWrapper stream: output stream of a running process.
+    :return: all the output that the running process generated up until that point.
+    :rtype: str
+    """
+
+    def enqueue_output(out, queue):
+        for line in iter(out.readline, b''):
+            queue.put(line)
+        out.close()
+
+    lines = []
+    q = Queue()
+    t = Thread(target=enqueue_output, args=(stream, q))
+    t.daemon = True
+    t.start()
+
+    while True:
```

Comment:
This should be discussed in #137. Reading output from a running process is blocking, and since there's no deterministic way to know when to stop reading, the read would end up hanging forever. I've actually tried many different solutions for non-blocking reading, and this was the only one that worked reliably.